src/app.py

```diff
@@ -194,6 +197,87 @@ def render_form(**kwargs):
                        new_url=new_url)
 
 
+@app.route("/pull_request_hook", methods=["POST"])
+def pull_request_hook():
+    event = request.headers.get('X-GitHub-Event')
+    if event == "ping":
+        return Response('Pong')
+    elif event != "pull_request":
+        return Response("Wrong event type", status=400)
+
+    payload = request.get_json()
+    action = payload['action']
+    if action not in ("opened", "edited", "reopened", "synchronize"):
+        return Response("Not Interested")
+    # status=204 : No Content
+
+    sender     = payload['sender']['login']
+
+    head_sha   = payload['pull_request']['head']['sha']
+    head_label = payload['pull_request']['head']['label']
+    head_ref   = payload['pull_request']['head']['ref']
+
+    base_sha   = payload['pull_request']['base']['sha']
+    base_label = payload['pull_request']['base']['label']
+    base_ref   = payload['pull_request']['base']['ref']
+
+    pull_num   = payload['pull_request']['number']
+    pull_url   = payload['pull_request']['html_url']
+
+    pull_ref   = "pull/{pull_num}/head".format(**locals())
+
+    # make sure data repo contains relevant commits
+    stdout, stderr, ret = fetch_data_ref(base_ref, pull_ref)
+
+    if ret == 0:
+        script = src_dir + "/tests/automerge_downtime_ok.py"
```

Comment:
Why invoke a subprocess instead of directly executing from the python interpreter?

Reply:
>Why

I like the idea of process separation -- let the check script do whatever it needs to do, and don't worry about the check having side effects on `app.py`'s python interpreter.  Eg, at some point in its development history, the check script called `chdir`, which i think should be fair game for a separate script to do.  Now the `chdir` is done via the `cwd` parameter passed to `subprocess.Popen`, which still happens in the subprocess.

Also, even the fact that the check script happens to be named with a `.py` extension, i consider incidental.  I'd rather not have to care at all what language it's implemented in.

And, why _not_? I don't really see any disadvantages, apart from the small bit of overhead of invoking a second python interpreter.